File: LibDialog/src/main/java/com/ooftf/master/widget/dialog/ui/ListDialog.java
package com.ooftf.master.widget.dialog.ui;

import android.app.Activity;

import com.ooftf.master.widget.dialog.delegate.ListDialogDelegate;
import com.ooftf.master.widget.dialog.inteface.DialogOnItemClickListener;
import com.ooftf.master.widget.dialog.inteface.ListDialogInterface;

import org.jetbrains.annotations.NotNull;

import java.util.List;

/**
 * @author ooftf
 * @email 994749769@qq.com
 * @date 2018/10/17 0017
 */
public class ListDialog extends BottomDialog implements ListDialogInterface {
    ListDialogDelegate dialogDelegate;

    public ListDialog(@NotNull Activity activity) {
        super(activity);
        dialogDelegate = new ListDialogDelegate(this, activity);
    }

    @Override
    public ListDialogInterface setShowCancel(boolean showCancel) {
        return dialogDelegate.setShowCancel(showCancel);
    }

    @Override
    public ListDialogInterface setList(List<String> data) {
        return dialogDelegate.setList(data);
    }

    @Override
    public ListDialogInterface setOnItemClickListener(DialogOnItemClickListener listener) {
        return dialogDelegate.setOnItemClickListener(listener);
    }

    @Override
    public ListDialogInterface show_() {
        dialogDelegate.show_();
        return this;
    }
}

File: lib-dialog/src/main/java/com/ooftf/master/widget/dialog/ui/ListDialog.kt
package com.ooftf.master.widget.dialog.ui

import android.app.Activity
import android.widget.TextView
import com.ooftf.master.widget.dialog.delegate.ListDialogDelegate
import com.ooftf.master.widget.dialog.inteface.DialogOnItemClickListener
import com.ooftf.master.widget.dialog.inteface.ListDialogInterface

/**
 * @author ooftf
 * @email 994749769@qq.com
 * @date 2018/10/17 0017
 */
open class ListDialog(activity: Activity) : BottomDialog(activity), ListDialogInterface {
    var dialogDelegate: ListDialogDelegate = ListDialogDelegate(this, activity)
    override fun setItemViewAdapter(adapter: ((textView: TextView, position: Int) -> Unit)): ListDialog {
        dialogDelegate.setItemViewAdapter(adapter)
        return this
    }

    override fun setShowCancel(showCancel: Boolean): ListDialog {
        dialogDelegate.setShowCancel(showCancel)
        return this
    }

    override fun setList(data: List<String>): ListDialog {
        dialogDelegate.setList(data)
        return this
    }

    override fun setOnItemClickListener(listener: DialogOnItemClickListener): ListDialog {
        dialogDelegate.setOnItemClickListener(listener)
        return this
    }

}